src/test/regress/sql/alter_table_ao.sql

```diff
@@ -447,3 +447,38 @@ CREATE table ao_insert_empty_row (a integer, b text, c integer) WITH (APPENDONLY
 INSERT INTO ao_insert_empty_row SELECT 1,'a',1 FROM gp_id WHERE dbid=-999;
 -- assert non-empty seg entries
 SELECT * FROM gp_toolkit.__gp_aoseg('ao_insert_empty_row'::regclass);
+
+-- test case: alter AO table add partition, the preference of the storage setting is: the encoding clause > table setting > gp_default_storage_options
+SET gp_add_partition_inherits_table_setting = on;
+
+CREATE TABLE ao_alter_add_part(a int, b int, c int) WITH (appendonly=true, orientation=row, compresstype=zstd, compresslevel=4, blocksize=65536) DISTRIBUTED BY (a)
+    PARTITION BY RANGE (b)
+        SUBPARTITION BY LIST (c) SUBPARTITION TEMPLATE (SUBPARTITION "11" VALUES (1, 2, 3) WITH (appendonly=true, orientation=row, compresstype=zstd, compresslevel=5, blocksize=32768))
+    (PARTITION "10" START (0) INCLUSIVE END (20) EXCLUSIVE);
+SET gp_default_storage_options = 'appendonly=true, orientation=row, compresstype=zlib, compresslevel=2';
+-- use statement encoding
+ALTER TABLE ao_alter_add_part ADD PARTITION "30" START (20) INCLUSIVE END (30) EXCLUSIVE WITH (appendonly=true, orientation=row, compresstype=zlib, compresslevel=3, blocksize=16384);
+ALTER TABLE ao_alter_add_part ADD PARTITION "60" START (50) INCLUSIVE END (60) EXCLUSIVE WITH (appendonly=true, orientation=row);
+-- use table setting
```

Comment:
may be settingS?

Reply:
it was copied [from](https://github.com/arenadata/gpdb/pull/618/files#diff-97261ee87e4c6b382b99dbdedad7d32d5f8ee80afd683cd5421d7a120e6ecd4cR482)